src/DynamoCore/Migration/Migration.cs

```diff
@@ -321,22 +321,16 @@ orderby result.From
             var migrationData = new NodeMigrationData(elNode.OwnerDocument);
             migrationData.AppendNode(elNode as XmlElement);
 
-            while (workspaceVersion != null && workspaceVersion < currentVersion)
+            foreach(var migration in migrations.Where(m=>m.From >= workspaceVersion))
```

Comment:
I'll remove `CurrentVersion`, and I propose to rename `from:` to `version`, as in "This node migration is applied to nodes in _version_ x.x.x and higher."